src/integrationTest/java/com/mongodb/hibernate/IdentifierIntegrationTests.java

```diff
@@ -0,0 +1,250 @@
+        static final String FIELD_NAME = "field with space Mixed Case";
+
+        @Id
+        int id;
+
+        @Column(name = WithSpaceAndDotMixedCase.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = InSingleQuotesMixedCase.COLLECTION_NAME)
+    static class InSingleQuotesMixedCase {
+        static final String COLLECTION_NAME = "`collection in single quotes Mixed Case`";
+        static final String FIELD_NAME = "`field in single quotes Mixed Case`";
+        static final String ACTUAL_COLLECTION_NAME = "collection in single quotes Mixed Case";
+        static final String ACTUAL_FIELD_NAME = "field in single quotes Mixed Case";
+
+        @Id
+        int id;
+
+        @Column(name = InSingleQuotesMixedCase.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = StartingWithSingleQuote.COLLECTION_NAME)
+    static class StartingWithSingleQuote {
+        static final String COLLECTION_NAME = "`collection starting with single quote";
+        static final String FIELD_NAME = "`field starting with single quote";
+
+        @Id
+        int id;
+
+        @Column(name = StartingWithSingleQuote.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = EndingWithSingleQuote.COLLECTION_NAME)
+    static class EndingWithSingleQuote {
+        static final String COLLECTION_NAME = "collection ending with single quote`";
+        static final String FIELD_NAME = "field ending with single quote`";
+
+        @Id
+        int id;
+
+        @Column(name = EndingWithSingleQuote.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = InDoubleQuotesMixedCase.COLLECTION_NAME)
+    static class InDoubleQuotesMixedCase {
+        static final String COLLECTION_NAME = "\"collection in double quotes Mixed Case\"";
+        static final String FIELD_NAME = "\"field in double quotes Mixed Case\"";
+        static final String ACTUAL_COLLECTION_NAME = "collection in double quotes Mixed Case";
+        static final String ACTUAL_FIELD_NAME = "field in double quotes Mixed Case";
+
+        @Id
+        int id;
+
+        @Column(name = InDoubleQuotesMixedCase.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = StartingWithDoubleQuote.COLLECTION_NAME)
+    static class StartingWithDoubleQuote {
+        static final String COLLECTION_NAME = "\"collection starting with double quote";
+        static final String FIELD_NAME = "\"field starting with double quote";
+
+        @Id
+        int id;
+
+        @Column(name = StartingWithDoubleQuote.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = EndingWithDoubleQuote.COLLECTION_NAME)
+    static class EndingWithDoubleQuote {
+        static final String COLLECTION_NAME = "collection ending with double quote\"";
+        static final String FIELD_NAME = "field ending with double quote\"";
+
+        @Id
+        int id;
+
+        @Column(name = EndingWithDoubleQuote.FIELD_NAME)
+        int v;
+    }
+}
```

Comment:
seems `EndingWithDoubleQuoteOnly` might be more accurate for the `MixedCase` ends in the same way as well, right?

Reply:
I don't think `EndingWithDoubleQuoteOnly` is better than `EndingWithDoubleQuote`. It is true that `EndingWithDoubleQuote` does not say explicitly that it does not start with a double quote, but `EndingWithDoubleQuoteOnly` does not say that either. Instead, it says that the name ends only with a double quote, as if there are other tests where it ends with not just a double quote, but with a double quote and something else.

To me it the meaning of `EndingWithDoubleQuote` seems quite clear, given that there is `StartingWithDoubleQuote` and `InDoubleQuotesMixedCase`.